lib/public-stats.js

```diff
@@ -274,3 +275,77 @@ const updateRetrievalTimings = async (pgClient, committees) => {
     JSON.stringify(rows)
   ])
 }
+
+/**
+ * @param {pg.Client} pgClient
+ * @param {Iterable<Committee>} committees
+ * @param {(minerId: string, cid: string) => (string[] | undefined)} findDealClients
+ */
+const updateDailyClientStats = async (pgClient, committees, findDealClients) => {
+  const clientRetrievalStats = new Map()
+  /** @type {Map<string, Array<Measurement>>} */
+  const clientToMeasurementsMap = new Map()
+  for (const c of committees) {
+    const { minerId, payload_cid: payloadCid } = c.retrievalTask
+
+    const clients = findDealClients(minerId, payloadCid)
+    if (!clients || !clients.length) {
+      console.warn(`Invalid retrieval task (${minerId}, ${payloadCid}): no deal clients found. Excluding the task from daily per-client stats.`)
+      Sentry.captureException(new Error('Invalid retrieval task: no deal clients found.'), {
+        extra: {
+          minerId,
+          payload_cid: payloadCid
+        }
+      })
+      continue
+    }
+    clients.forEach(clientId => {
+      if (clientToMeasurementsMap.has(clientId)) {
+        clientToMeasurementsMap.get(clientId).push(...c.measurements)
+      } else {
+        clientToMeasurementsMap.set(clientId, Array.from(c.measurements))
+      }
+    })
+  }
+
+  clientToMeasurementsMap.forEach((measurements, clientId) => {
```

Comment:
```suggestion
  for (const [clientId, measurements] of Object.entries(clientToMeasurementsMap)) {
```

See https://stackoverflow.com/a/54550693